tests/e2e/test_read.py

```diff
@@ -125,5 +99,8 @@ def test_random_seeks_and_reads(self, small_blob, n):
     def test_read_using_iter(self, small_with_newlines_blob):
         with BlobIO(small_with_newlines_blob.url, "rb") as f:
             lines = [line for line in f]
-            expected_lines = small_with_newlines_blob.data.splitlines(keepends=True)
+            expected_lines = [
+                line + b"\n" for line in small_with_newlines_blob.data.split(b"\n")[:-1]
+            ]
+            expected_lines.append(small_with_newlines_blob.data.split(b"\n")[-1])
```

Comment:
For fixing this test, let's just wrap the expected data with a `io.BytesIO` object to get the expected lines e.g.:
```python
 expected_lines = io.BytesIO(small_with_newlines_blob.data).readlines()
```
Mainly this helps make it less verbose and easier to follow what we are expecting.